plugin/core/tree_view.py

```diff
@@ -0,0 +1,269 @@
+from .css import css
+from .promise import Promise
+from .typing import Dict, IntEnum, List, Optional, TypeVar
+from .views import SublimeKind
+from abc import ABCMeta
+from abc import abstractmethod
+from functools import partial
+import html
+import sublime
+import uuid
+
+
+T = TypeVar('T')
+
+
+class TreeItemCollapsibleState(IntEnum):
+    COLLAPSED = 1
+    EXPANDED = 2
+
+
+class TreeItem:
+
+    def __init__(
+        self,
+        label: str,
+        kind: SublimeKind = sublime.KIND_AMBIGUOUS,
+        description: str = "",
+        tooltip: str = "",
+        command_url: str = ""
+    ) -> None:
+        self.label = label
+        """ A human-readable string describing this item. """
+        self.kind = kind
+        """ A sublime.Kind tuple which is rendered as icon in front of the label. """
+        self.description = description
+        """ A human-readable string which is rendered less prominent. """
+        self.tooltip = tooltip
+        """ The tooltip text when you hover over this item. """
+        self.command_url = command_url
+        """ A HTML embeddable URL for a command that should be executed when the tree item label is clicked.
+        Use the sublime.command_url function to generate this URL. """
+        self.collapsible_state = TreeItemCollapsibleState.COLLAPSED
+        self.id = str(uuid.uuid4())
+
+    def html(self, sheet_name: str, indent_level: int) -> str:
+        indent_html = '<span style="padding-left: {}rem;">&nbsp;</span>'.format(indent_level)
+        if self.collapsible_state == TreeItemCollapsibleState.COLLAPSED:
+            disclosure_button_html = '<a class="disclosure-button" href="{}">&#9654;</a>'.format(
+                sublime.command_url('lsp_expand_tree_item', {'name': sheet_name, 'id': self.id}))
+        elif self.collapsible_state == TreeItemCollapsibleState.EXPANDED:
+            disclosure_button_html = '<a class="disclosure-button" href="{}">&#9660;</a>'.format(
```

Comment:
Sorry for nitpcking but we could likely just paste an actual  utf8 characters instead of using entities. Entities are less readable for the developer and I don't think there is really a benefit in using entities (as a web dev I should know but frankly I don't really ;)).

Reply:
The reason I tend to avoid uncommon unicode characters if there is an ASCII equivalent, is that often they mess up the horizontal alignment in the source code. For example if you have `"rulers": [120]` set for Python, then the ruler is not reliable anymore for such lines (and often I don't know which unicode characters count as 1 or as 2 character-width).

![emoji](https://user-images.githubusercontent.com/6579999/209663452-328b6a2e-458b-4ae5-a822-6d0eee1a169e.gif)

But here I can change it, no prob

Reply:
I replaced the arrow symbols, but left the entity in place of the ⇄ symbol for now. Because that one is exactly such a case with the implication from the GIF (on my OS and system font).